src/client/visualizations/base-visualization/base-visualization.tsx

```diff
@@ -0,0 +1,223 @@
+import * as React from 'react';
+import { $, ply, TimeRange, Expression, Dataset } from 'plywood';
+import {
+  MeasureModeNeeded,
+  Manifest,
+  Measure,
+  VisualizationProps,
+  DatasetLoad,
+  Splits,
+  DataSource,
+  Colors,
+  Resolve,
+  Essence
+} from '../../../common/models/index';
+
+import { SPLIT, SEGMENT, TIME_SEGMENT } from '../../config/constants';
+
+
+export interface BaseVisualizationState {
+  datasetLoad?: DatasetLoad;
+  dragStartTime?: Date;
+  dragTimeRange?: TimeRange;
+  roundDragTimeRange?: TimeRange;
+  dragOnMeasure?: Measure;
+  scrollLeft?: number;
+  scrollTop?: number;
+  hoverTimeRange?: TimeRange;
+  hoverMeasure?: Measure;
+}
+
+export class BaseVisualization<S extends BaseVisualizationState>
+  extends React.Component<VisualizationProps, S> {
+
+  public static id = 'base-visualization';
+  public static title = 'Base Visualization';
+
+  public static measureModeNeed: MeasureModeNeeded = 'single';
+
+  public static handleCircumstance(
+    dataSource: DataSource, splits: Splits, colors: Colors, current: boolean
+  ): Resolve {
+    return Resolve.NEVER;
+  }
+
+  // isMounted was already taken by the superclass
+  protected _isMounted: boolean;
+
+  constructor() {
+    super();
+
+    this.state = this.getDefaultState() as S;
+
+    this.globalMouseMoveListener = this.globalMouseMoveListener.bind(this);
+    this.globalMouseUpListener = this.globalMouseUpListener.bind(this);
+    this.globalKeyDownListener = this.globalKeyDownListener.bind(this);
+  }
+
+  getDefaultState(): BaseVisualizationState {
+    return {
```

Comment:
Yeah, but I think I didn't use `getInitialState` for typing reasons (the transpiler was complaining).